src/app/Http/Controllers/Auth/LoginController.php

```diff
@@ -0,0 +1,130 @@
+{
+    public function index(): View
+    {
+        $view = config('user-login.view');
+
+        $action = route('login');
+
+        $buttonLabel = 'Sign in';
+
+        $questions = [
+            [
+                'name' => 'username',
+                'label' => 'What is your username?',
+                'hint' => 'The username you use to access your Windows device, such as jdoe3.',
+            ],
+            [
+                'name' => 'password',
+                'label' => 'What is your password?',
+                'hint' => 'The password you use to access your Windows device.',
+            ],
+        ];
+
+        return $view === 'gov-uk-login'
+            ? GovukPage::questions(
+                'Sign in',
+                [
+                    GovukQuestion::input($questions[0]['label'], $questions[0]['name'])
+                        ->hint($questions[0]['hint'])
+                        ->width(20),
+
+                    GovukQuestion::input($questions[1]['label'], $questions[1]['name'])
+                        ->hint($questions[1]['hint'])
+                        ->width(20),
+                ],
+                $buttonLabel,
+                $action,
+                null,
+                'post',
+                "user-login::$view"
+            )
+            : view("user-login::$view", [
+                'questions' => $questions,
+                'action' => $action,
+                'buttonLabel' => $buttonLabel,
+            ]);
+    }
+
+    public function signIn(LoginRequest $request): RedirectResponse
+    {
+        $usernameKey = config('user-login.ldap-sync') === true ? 'samaccountname' : 'username';
+
+        $details = [
+            $usernameKey => strtolower($request->username),
+            'password' => $request->password,
+        ];
+
+        if ($this->syncExistingUser($details[$usernameKey]) === false) {
+            return $this->loginFailed($details[$usernameKey]);
+        }
+
+        return Auth::attempt($details, true) === true
+            ? $this->loginSucceeded()
+            : $this->loginFailed($details[$usernameKey]);
+    }
+
+    protected function syncExistingUser(string $username): bool
+    {
+        $model = config('user-login.model');
+
+        $mail = $model::mail($username);
+
+        if ($mail === null) {
+            return false;
+        }
+
+        $model::query()
+            ->where('username', '=', $username)
+            ->orWhere('email', '=', $mail)
+            ->limit(1)
+            ->update([
+                'username' => $username,
+                'email' => $mail,
+            ]);
+
+        return true;
+    }
+
+    public function signOut(): RedirectResponse
+    {
+        Auth::logout();
+
+        return redirect()->route('login');
+    }
+
+    protected function loginFailed(string $username): RedirectResponse
+    {
+        flash('Sign in failed; please check your username and password and try again')->error();
```

Comment:
"Sign-in failed; check your details and try again."

Could be configurable message.

Reply:
Should be in error part of the function.

Reply:
fixed